pkg/generator/main.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	apiext "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/controller-tools/pkg/crd"
+	"sigs.k8s.io/controller-tools/pkg/loader"
+	"sigs.k8s.io/controller-tools/pkg/markers"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	ExperimentalMarker      = "<gateway:experimental>"
```

Comment:
If you use `+gateway:experimental` here, does the parser remove it from the description of the field in the JSON schema?

What I'm thinking is if we can use the same string pattern as tags, while using the grep logic you have implemented below. It looks and _feels_ more readable.

Reply:
>If you use +gateway:experimental here, does the parser remove it from the description of the field in the JSON schema?

Yep, unfortunately those get pulled out as "markers". Unfortunately I have not found a good way to access those. I tried to add support for a custom marker like this (https://github.com/kubernetes-sigs/controller-tools/pull/635), but it ended up being complex to get it working at every level I wanted (and represented significantly more code and complexity). If we ever want to upstream the concept of experimental fields, that's likely what we'd do, but I think that's something we can save for later.